### PR TITLE
feat: Screen wake lock feature

### PR DESCRIPTION
**Pull request by Arthurk12**
### What does this PR do?

This PR adds the screen wake lock feature, which is available only for mobile users using browser with support to the given API.
When an user with a supported device joins the meeting, a toast is displayed offering to activate the feature.
Adds a toggle under settings menu to activate/deactivate it.

New settings.yml flag `wakeLock.enabled` default is `false`
wake_lock_feature

### CODE REVIEW

File: bigbluebutton-html5/private/config/settings.yml [hunk 1]
```diff
@@ -59,6 +59,7 @@ public:
     # the default logoutUrl matches window.location.origin i.e. bigbluebutton.org for demo.bigbluebutton.org
     # in some cases we want only custom logoutUrl to be used when provided on meeting create. Default value: true
     askForConfirmationOnLeave: true
+    enableWakeLock: false
```
Comment (antobinary) [inline bigbluebutton-html5/private/config/settings.yml:62]: Should we leave only one of the configs?

File: bigbluebutton-html5/imports/ui/components/wake-lock/component.jsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  getToast(id, title, button1, button2) {
+    const { intl } = this.props;
+
+    return (
+      <div id={id}>
+        <Styled.Title>
+          { intl.formatMessage(intlMessages[title]) }
+        </Styled.Title>
+        <Styled.ToastButtons>
+          { button1 }
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/wake-lock/component.jsx:93]: Do we have more descriptive names?

File: bigbluebutton-html5/imports/ui/components/wake-lock/component.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      success ? 'wakeLockAcquireSuccess' : 'wakeLockAcquireFailed', null, null);
+    setTimeout(() => {
+      notify(feedbackToast, success ? 'success' : 'error', 'lock', feedbackToastProps, null, true);
+    }, 800);
+  }
+
+  render() {
+    const { wakeLockSettings, request, release } = this.props;
+    if (wakeLockSettings) {
+      request().then((success) => {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/wake-lock/component.jsx:117]: Is `success` here acting like an `error`?

File: bigbluebutton-html5/imports/ui/components/wake-lock/service.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  release() {
+    if (this.isActive()) this.sentinel.release();
+  }
+}
+
+const wakeLock = new WakeLock();
+
+export default {
+  isEnabled: () => WakeLock.isEnabled(),
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/wake-lock/service.js:70]:
```suggestion
  isEnabled: () => wakeLock.isEnabled(),
```

**Review by antobinary — COMMENTED** [review]
Great addition, thank you @Arthurk12 !

I left a few comments.

File: bigbluebutton-html5/private/config/settings.yml (hunk 1 above)
Comment (antobinary) [reply bigbluebutton-html5/private/config/settings.yml:62]: For whether the feature is enabled or not, let's mimic how other [recent] features do it -- see darkMode for example,

```
featureName
  enabled: Boolean
```

**Review by ramonlsouza — CHANGES_REQUESTED** [review]
If I choose "Yes!" it says that the feature is active, but in settings it appears as OFF.

https://github.com/bigbluebutton/bigbluebutton/assets/3728706/45e16c1d-5ed2-4d86-a8fc-459fe4c34025

**Review by ramonlsouza — APPROVED** [review]

**Pull request merged**